dashboard/app/controllers/script_levels_controller.rb

```diff
@@ -420,4 +430,23 @@ def present_level
   def protect_against_forgery?
     return false
   end
+
+  def redirect_script(level, locale)
+    # Return nil if level is nil level does not belong to a script.
+    return nil if !level || level.script.nil?
+
+    # Return nil we know the user can view the script version requested.
+    script = level.script
```

Comment:
I think the section above would read a little better as:
```ruby
script = level&.script
return nil unless script
```
but if that seems backwards to you for some reason, then don't worry about it :)

Reply:
thanks! your suggestion is definitely more straightforward